core/trino-spi/src/main/java/io/trino/spi/connector/DynamicFilter.java

```diff
@@ -81,4 +88,12 @@ public TupleDomain<ColumnHandle> getCurrentPredicate()
     boolean isAwaitable();
 
     TupleDomain<ColumnHandle> getCurrentPredicate();
+
+    /**
+     * Returns preferred timeout in milliseconds if build side can be estimated otherwise empty, that connector should wait
+     * for the dynamic filter to be narrowed down since split enumeration started.
+     * Future from {@link DynamicFilter#isBlocked()} method should be acquired before getting preferred dynamic filter timeout.
```

Comment:
@Dith3r could you remind me why it was the case?

Reply:
If you call `getPreferredTimeout` before acquiring future from `isBlocked` you can receive future with excluded already resolved DF associated with preferred timeout and wait for DFs that have no preferred timeout set.

Reply:
If DF is already resolved, then no waiting will happen anyway, right?

Reply:
Yes.